contract/contracts/selling-lock/src/entry.rs

```diff
@@ -1,39 +1,121 @@
+    u64::from_le_bytes(buf)
+}
 
-    // return an error if args is invalid
-    if args.is_empty() {
-        return Err(Error::MyError);
-    }
+fn outputs_contains_owner_cell(args: &Bytes) -> Result<bool, Error> {
+    let contains_owner_cell = QueryIter::new(load_cell, Source::Output)
+        .find(|output: &CellOutput| {
+            debug!("outputs_contains_owner_cell:");
+            selling_lock_args_same_as_script(args, &output.lock())
+        }).is_some();
+    Ok(contains_owner_cell)
+}
+
+fn collect_outputs_owner_amount(args: &Bytes) -> Result<u64, Error> {
+    debug!("enter collect_outputs_owner_amount:");
+    let mut buf = [0u8; MINIMAL_CAPACITY_LEN];
+    let capacity_list = QueryIter::new(load_cell, Source::Output)
+        .map(|cell: CellOutput|{
+            debug!("now collect_outputs_owner_amount:");
+            debug!("selling_lock_args_same_as_script: {:?}",selling_lock_args_same_as_script(args, &cell.lock()));
+            debug!("cell.type_().as_reader().is_none(): {:?}",cell.type_().as_reader().is_none());
+            if selling_lock_args_same_as_script(args, &cell.lock()) && cell.type_().as_reader().is_none() {
+                debug!("&cell.capacity().raw_data(): {:?}",&cell.capacity().raw_data());
+                buf.copy_from_slice(&cell.capacity().raw_data());
+                return Ok(u64::from_le_bytes(buf));
+            } else {
+                return Ok(0u64)
+            }
+        }).collect::<Result<Vec<_>, Error>>()?;
+    Ok(capacity_list.into_iter().sum::<u64>())
+}
 
-    let tx_hash = load_tx_hash()?;
-    debug!("tx hash is {:?}", tx_hash);
+fn get_price(data: &Bytes) -> u64 {
+    let data_len = data.len();
+    let start_point = data_len - MINIMAL_CAPACITY_LEN;
+    let mut buf = [0u8; MINIMAL_CAPACITY_LEN];
+    buf.copy_from_slice(&data[start_point..data_len]);
+    u64::from_le_bytes(buf)
+}
+pub fn main() -> Result<(), Error> {
+    let script = load_script()?;
+    let args: Bytes = script.args().unpack();
+    debug!("script args is {:?}", &args);
+    let mut buf = [0u8; 32];
+    buf.copy_from_slice(&args.as_ref()[0..32]);
+    debug!("code hash is {:?}", buf);
 
-    let _buf: Vec<_> = vec![0u8; 32];
+    let self_capacity = get_self_capacity(&args);
+    debug!("self capacity is {:?}", self_capacity);
 
-    Ok(())
+    if check_is_owner(&args)? {
+        debug!("unlock by owner!");
+        return Ok(());
+    } else{
+        debug!("unlock by purchase!");
+        /*
+         * outputs.contains(owner_lock) && 
+         * output_owner_cell.capacity >= minimal_capaicty + self.capacity && 
+         * output_owner_cell.type_script == null
+         */
+        let sell_price = get_price(&args);
+        let paid_price = collect_outputs_owner_amount(&args)?;
+        debug!("sell price is {:?}", sell_price);
+        debug!("paid price is {:?}", paid_price);
+        if outputs_contains_owner_cell(&args)? && paid_price >= sell_price + self_capacity {
```

Comment:
To prevent a malicious type script validator, the output owner cell type must be empty

Reply:
added logic to ensure there is no type scipt in output owner cell